components/learning/PublicContent.vue

```diff
@@ -10,14 +10,16 @@
     <div class="public-course__units-list">
       <div
         v-for="unit of course.units"
-        :key="unit"
+        :key="unit.id"
         class="public-course__units-item"
       >
         <ProgressCircle :progress="0" />
         <BasicLink
-          :to="`/course/${course.slug}/${unitStore.unitById(unit).value?.slug}`"
+          :to="`/course/${course.slug}/${
+            unitStore.unitById(unit.id).value?.slug
+          }`"
```

Comment:
```suggestion
          :to="`/course/${course.slug}/${unitStore.unitById(unit.id).value!.slug}`"
```

I think we should assume that the unit exits (with the `!` instead of `?`).

And if not, we should control what to render if there is no unit.
But I believe that it cannot be the case that there is no unit here.

Reply:
Same as before :)

Reply:
Ok!